File: memory-cache/typescript/tests/FixedClock.ts
import type { Clock } from '../src/Clock.js';

export class FixedClock implements Clock {
  private current: Date;

  constructor(start: Date) {
    this.current = start;
  }

  now(): Date {
    return this.current;
  }

  advanceTo(when: Date): void {
    this.current = when;
  }

  advanceMs(ms: number): void {
    this.current = new Date(this.current.getTime() + ms);
  }
}

File: memory-cache/python/tests/fixed_clock.py
from __future__ import annotations

from datetime import datetime, timedelta


class FixedClock:
    def __init__(self, now: datetime) -> None:
        self._now = now

    def now(self) -> datetime:
        return self._now

    def advance_to(self, when: datetime) -> None:
        self._now = when

    def advance(self, delta: timedelta) -> None:
        self._now = self._now + delta
